### PR TITLE
Gracefully handle previous version install failure in upgrade test.

### PR DESCRIPTION
**Pull request by igorpeshansky**
Before the initial release on a given platform, this test is effectively guaranteed to fail, because there is no stable package to install yet. So we skip this test if the pre-upgrade version fails to install. Since `setupOpsAgentFrom` will retry a few times, the the test should only be skipped if all retries fail consistently.

### CODE REVIEW

**Review by alison0211 — APPROVED** [review]

File: integration_test/ops_agent_test.go [hunk 1]
```diff
@@ -1425,6 +1425,10 @@ func TestUpgradeOpsAgent(t *testing.T) {
 		// a default value of "", which means stable.
 		firstVersion := packageLocation{repoSuffix: os.Getenv("REPO_SUFFIX_PREVIOUS")}
 		if err := setupOpsAgentFrom(ctx, logger, vm, "", firstVersion); err != nil {
+			// Installation from stable may fail before the first release.
+			if firstVersion.repoSuffix == "" && (strings.HasPrefix(err.Error(), "installOpsAgent() failed to run googet") || strings.HasPrefix(err.Error(), "installOpsAgent() error running repo script")) {
+				return
```
[the remark this answers is not in the document]
Comment (igorpeshansky) [reply integration_test/ops_agent_test.go]: Actually, I should've used `t.Skip()` anyway, which would also log. Done.

**Review by Subbarker — APPROVED** [review]
LGTM

File: integration_test/ops_agent_test.go [hunk 2]
```diff
@@ -1425,6 +1425,10 @@ func TestUpgradeOpsAgent(t *testing.T) {
 		// a default value of "", which means stable.
 		firstVersion := packageLocation{repoSuffix: os.Getenv("REPO_SUFFIX_PREVIOUS")}
 		if err := setupOpsAgentFrom(ctx, logger, vm, "", firstVersion); err != nil {
+			// Installation from stable may fail before the first release.
+			if firstVersion.repoSuffix == "" && (strings.HasPrefix(err.Error(), "installOpsAgent() failed to run googet") || strings.HasPrefix(err.Error(), "installOpsAgent() error running repo script")) {
+				t.Skip(fmt.Printf("Installing stable agent failed with error %v; assuming first release.", err))
```
Comment (martijnvans) [inline integration_test/ops_agent_test.go:1430]: there's a t.Skipf function that does the fmt stuff for you.

**Review by martijnvans — APPROVED** [review]

File: integration_test/ops_agent_test.go (hunk 2 above)
Comment (igorpeshansky) [reply integration_test/ops_agent_test.go:1430]: TIL. Done.

**Pull request merged**